core/server/data/xml/sitemap/base-generator.js

```diff
@@ -55,13 +55,18 @@ _.extend(BaseSiteMapGenerator.prototype, {
         // Create all the url elements in JSON
         var self = this,
             nodes;
-        nodes = _.map(data, function (datum) {
+
+        nodes = _.reduce(data, function (nodeArr, datum) {
```

Comment:
If you prefer we can not add the filters to the getData call, and just rely on the validations.